docs/guides/1.3.0-upgrade-guide.md

```diff
@@ -0,0 +1,137 @@
+description: |-
+Couchbase Capella Provider 1.3.0: Upgrade and Information Guide
+---
+
+# Couchbase Capella Provider 1.3.0: Upgrade and Information Guide
+
+Here is a list of what's new in 1.3.0
+
+## New Features
+
+* Enable Network Peer [`couchbase-capella_network_peer`](https://registry.terraform.io/providers/couchbasecloud/couchbase-capella/latest/docs/resources/network_peer)
+* Enable Private Endpoint Service [`couchbase-capella_private_endpoint_service`](https://registry.terraform.io/providers/couchbasecloud/couchbase-capella/latest/docs/resources/private_endpoint_service)
+* Accept or Reject Private Endpoints [`couchbase-capella_private_endpoints`](https://registry.terraform.io/providers/couchbasecloud/couchbase-capella/latest/docs/resources/private_endpoints)
+
+## Changes
+
+There are no deprecations as part of this release.
+
+## Network Peer
+
+To create a new network peer use the `network-peer` resource as follows:
+```
+resource "couchbase-capella_network_peer" "new_network_peer" {
+  organization_id = var.organization_id
+  project_id      = var.project_id
+  cluster_id      = var.cluster_id
+  name            = var.network_peer.name
+  provider_type   = var.network_peer.provider_type
+  provider_config = {
+    aws_config = {
+      account_id = var.aws_config.account_id
+      vpc_id     = var.aws_config.vpc_id
+      cidr       = var.aws_config.cidr
+      region     = var.aws_config.region
+    }
+  }
+}
+```
+```
+output "peer_id" {
+  value = couchbase-capella_network_peer.new_network_peer.id
+}
+```
+```
+output "new_network_peer" {
+  value = couchbase-capella_network_peer.new_network_peer
+}
+```
+For more information, see the [network peer examples](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/tree/main/examples/network_peer)
+
+## System Events
+
+Use the `event` data source to fetch a specific event.
+```
+data "couchbase-capella_event" "existing_event" {
+  organization_id = var.organization_id
+  id              = var.event_id
+}
+```
+```
+output "existing_event" {
+  value = data.couchbase-capella_event.existing_event
+}
+```
+Use the `events` data source to fetch existing list of events.
+```
+data "couchbase-capella_events" "existing_events" {
+  organization_id = var.organization_id
+}
+```
+```
+output "existing_events" {
+  value = data.couchbase-capella_events.existing_events
+}
+```
+
+## Private Endpoints
+
+Use the `aws_private_endpoint_command` data source to get AWS command for configuring VPC endpoint.
+```
+data "couchbase-capella_aws_private_endpoint_command" "aws_command" {
+  organization_id = var.organization_id
+  project_id      = var.project_id
+  cluster_id      = var.cluster_id
+  vpc_id          = var.vpc_id
+  subnet_ids      = var.subnet_ids
+}
+```
+```
+output "aws_command" {
+  value = data.couchbase-capella_aws_private_endpoint_command.aws_command
+}
+```
+Use the `azure_private_endpoint_command` data source to get AWS command for configuring VPC endpoint.
```

Comment:
@PaulomeeCb  Curious: I wonder how one will use this functionality in an automated environment with Terraform. The command needs to be run out of band before the rest of scripts can be executed. Should we provide some guidance here?

Reply:
@rajagp I didn't work on the private endpoints actually, @l0n3star would be able to address this better.

Reply:
@rajagp Good question.  Short answer, this command is not particularly useful in an automated environment.  

Long answer.  Assume customers will configure VPC endpoints with terraform as well.  They would have to use the CSP's terraform provider.  For example with AWS it might look like this:

```
resource "aws_vpc_endpoint" "example" {
  vpc_id            = var.vpc_id
  service_name      = "com.amazonaws.vpce.us-east-1.vpce-svc-1234"
  vpc_endpoint_type = "Interface"
  subnet_ids        = var.subnet_ids
}
```

Customers will have to supply `vpc_id` and `subnet_ids`.   `vpc_endpoint_type` must be `Interface`.  What customers need from us is `service_name`.  It would be better if we just returned the service name so it can be referenced in AWS's resource block.

Reply:
@l0n3star I think we should add an enhancement for 1.3.1 to return the `service_name`- we can do that in provider right?

Reply:
We can do it in the provider.  Just need to extract `service_name` from what's returned by the API.  

And of course need to do something similar for Azure.

Reply:
@rajagp 

On second thought I think regex is a bad idea.  What if the script changes and this breaks regex ?  A robust solution is to just return service name from the endpoint.  Luckily this is available to us from `network` component, so pretty easy to change V4 API.